### PR TITLE
[examples] Update `remix` template

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Bring in line with latest template

### CODE REVIEW

**Review by TooTallNate — APPROVED** [review]

File: examples/remix/app/root.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import type { MetaFunction } from "@vercel/remix";
+import { cssBundleHref } from "@remix-run/css-bundle";
+import type { LinksFunction } from "@remix-run/node";
```
Comment (TooTallNate) [inline examples/remix/app/root.tsx]: All instances of `@remix-run/node` should be replaced with `@vercel/remix`.

File: examples/remix/app/root.tsx [hunk 2]
```diff
@@ -1,4 +1,5 @@
-import type { MetaFunction } from "@vercel/remix";
+import { cssBundleHref } from "@remix-run/css-bundle";
```
Comment (TooTallNate) [inline examples/remix/app/root.tsx:1]: I tried to build this and got an error:

```
✘ [ERROR] Could not resolve "@remix-run/css-bundle"

    app/root.tsx:1:30:
      1 │ import { cssBundleHref } from "@remix-run/css-bundle";
        ╵                               ~~~~~~~~~~~~~~~~~~~~~~~
```

Is there a future flag we need to enable or something?

**Review by TooTallNate — CHANGES_REQUESTED** [review]

File: examples/remix/app/root.tsx (hunk 2 above)
Comment (MichaelDeBoey) [reply examples/remix/app/root.tsx:1]: I forgot to include the dependency 🙈

File: examples/remix/app/root.tsx (hunk 1 above)
Comment (MichaelDeBoey) [reply examples/remix/app/root.tsx]: Is there a specific reason for this?

Just trying to understand the reasoning behind it

**Comment by MichaelDeBoey** [comment]
@TooTallNate Is there anything I still need to do in order to get this one merged?

**Comment by MichaelDeBoey** [comment]
@leerob @TooTallNate Is there anything I still need to do in order to get this one merged?

File: examples/remix/package.json [hunk 3]
```diff
@@ -3,24 +3,28 @@
   "sideEffects": false,
   "scripts": {
     "build": "remix build",
-    "dev": "remix dev"
+    "dev": "remix dev",
```
Comment (leerob) [inline examples/remix/package.json:6]: Should we use the new dev server?

**Review by leerob — APPROVED** [review]
LGTM!

**Review by TooTallNate — APPROVED** [review]

File: examples/remix/app/root.tsx (hunk 1 above)
Comment (TooTallNate) [reply examples/remix/app/root.tsx]: This package ensures compatibility with both Vercel Serverless and Edge functions. You can read the docs [here](https://vercel.com/docs/frameworks/remix#@vercel/remix).

File: examples/remix/package.json (hunk 3 above)
Comment (MichaelDeBoey) [reply examples/remix/package.json:6]: @leerob As you can see in https://github.com/remix-run/remix/pull/6650, the only thing that changed for `v2_dev` is enabling the future flag

**Comment by MichaelDeBoey** [comment]
@leerob @TooTallNate It seems like the deploy isn't authorized and the `Enforce Changeset` lint is also failing
Anything I can do to make them work?

I'd love to get this one merged, so we can start pointing to the official template instead in the CLI

**Comment by TooTallNate** [comment]
No changeset necessary since this only updates an example. I'm just doing some final verification and will get this merged ASAP.

**Comment by TooTallNate** [comment]
Thank you @MichaelDeBoey!

**Pull request merged**